pkg/tracing/meta/notifications.go

```diff
@@ -0,0 +1,46 @@
+package meta
+
+import (
+	"sync"
+)
+
+var (
+	spanNotifications = make(map[string]chan error)
```

Comment:
i have some questions on how well this will scale as one executor many thousands of executions and spans at once, especially with eg. parallel steps that may each present 1K steps for spans also.  this may quickly balloon.  do we need this?  is it okay to say that as soon as we publish, it gets written to a buffer and sent to eg. kafka?

Reply:
Mm this is to ensure that it's part of the critical flow -  we need to know that it's been written to Kafka (and not just the buffer) before we're happy to continue.

Without this, there's always the risk that we write to the buffer and then fail to publish, resulting in lost data and a trace that appears as if something was missed.

Intentionally left this as a separate PR as I figured it would create discussion! 😄